tools/start_integration_tests.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+"""
+Start integration test jobs for PRs by non-team members that are approved by team members.
+"""
+
+import argparse
+import json
+import subprocess
+import sys
+from pathlib import Path
+import re
+
+
+CLI_REPO = "databricks/cli"
+CODE_OWNERS = "pietern andrewnester shreyas-goenka denik anton-107".split()
+
+
+def run(cmd):
+    sys.stderr.write("+ " + " ".join(cmd) + "\n")
+    return subprocess.run(cmd, check=True)
+
+
+def run_json(cmd):
+    sys.stderr.write("+ " + " ".join(cmd) + "\n")
+    result = subprocess.run(cmd, stdout=subprocess.PIPE, encoding="utf-8", check=True)
+
+    try:
+        return json.loads(result.stdout)
+    except Exception:
+        sys.stderr.write(f"Failed to JSON parse:\n{result.stdout}\n")
+        raise
+
+
+def get_approved_prs_by_non_team():
+    prs = run_json(["gh", "pr", "-R", CLI_REPO, "list", "--limit", "300", "--json", "number,author,reviews,headRefOid"])
+    result = []
+
+    for pr in prs:
+        pr_number = pr["number"]
+        author = pr["author"]["login"]
+
+        if author in CODE_OWNERS:
+            continue
+
+        approved_by = []
+        for review in pr.get("reviews", []):
+            approver = review["author"]["login"]
+            if review["state"] == "APPROVED" and approver in CODE_OWNERS:
+                approved_by.append(approver)
+
+        if not approved_by:
+            continue
+
+        result.append(
+            {
+                "number": pr_number,
+                "commit": pr["headRefOid"],
+                "author": author,
+                "approved_by": approved_by,
+            }
+        )
+
+    return prs, result
+
+
+def start_job(pr_number, commit_sha, author, approved_by, workflow, repo, force=False):
+    pr_details = run_json(["gh", "pr", "-R", CLI_REPO, "view", str(pr_number), "--json", "title,url"])
+    pr_title = pr_details.get("title", "")
+    commit_url = f"https://github.com/{CLI_REPO}/pull/{pr_number}/commits/{commit_sha}"
+    approvers = ", ".join(approved_by)
```

Comment:
This should filter on PRs made from other branches on the same repo, not forks.

As is, the following can happen:
* PR from fork branch is submitted
* Someone approves PR
* New (malicious) code is pushed to fork branch
* Automatically runs

For dependabot specifically, we know the PR is made from a branch in this repo.

Reply:
Done in f6ba22b01b0297f5ffbd8077430639fd0e238694

```
~/work/cli-main % ./tools/start_integration_tests.py -R databricks-eng/eng-dev-ecosystem
+ gh pr -R databricks/cli list --limit 300 --json number,author,reviews,headRefOid,headRepository,headRepositoryOwner
#2507 by kanterov skipped due to headRepository: {'id': 'R_kgDOJ4V9mw', 'name': 'cli'}
#1737 by RicNord skipped due to headRepository: {'id': 'R_kgDOMnDASw', 'name': 'cli-databricks'}
#1193 by ohbus skipped due to headRepository: {'id': 'R_kgDOLQLKbQ', 'name': 'databricks-cli'}
Fetched 52 PRs.
```